### ISSUE TITLE
Rename ignores conflicts from invalid nameof() string literals

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#1270]
(evil corner case issue)

Source:

```C#
using System.Runtime.InteropServices;

[Guid("439934D1-B051-4DC1-9303-1A91120EA1C" + nameof(A))]
class A { }
```

Renaming `A` to any name longer than one character should report an unresolvable conflict.

Optionally (this is not necessarily a good idea, and it certainly not worth doing), it can replace the `nameof` with the old name (under the assumption that GUIDs should not change)

**Comment by Pilchie** [comment]
I don't think it's worth addressing this corner case.

**Comment by SLaks** [comment]
No argument there :)
